easybuild/framework/easyconfig/types.py

```diff
@@ -400,6 +400,12 @@ def to_dependencies(dep_list):
     """
     return [to_dependency(dep) for dep in dep_list]
 
+def to_checksums(checksums):
+    if not any(isinstance(checksum, list) for checksum in checksums):
+        return to_list_of_strings_and_tuples(checksums)
+    else:
+        return [to_list_of_strings_and_tuples(cs) for cs in checksums]
```

Comment:
positive logic (and a docstring) makes this easier to comprehend:

```python
def to_checksums(checksums):
    """Ensure correct element types for list of checksums: convert each element to a tuple."""
    if all(isinstance(cs, basestring) for cs in checksums):
        # if all elements are strings, we convert each individual element to a tuple
        res = [to_list_of_strings_and_tuples(cs) for cs in checksums]
    else:
        res = to_list_of_strings_and_tuples(checksums)

    return res
```

(do check my comment/docstring, I may confuse between list/tuple...)